### PR TITLE
Merlin lsp opam file changes

### PR DESCRIPTION
**Pull request by rgrinberg**
* Add dependency on merlin
* Dune isn't a just build dependency

### CODE REVIEW

**Comment by andreypopp** [comment]
I don’t think a dependency on merlin is needed (merlin doesn't install any libs). Why dune isn’t just a build dependency?

**Comment by rgrinberg** [comment]
Oh okay. I removed merlin as a dependency.
> Why dune isn’t just a build dependency?
it is a build dependency, in the sense that it’s only required at build time. However, the way opam inteprets build dependencies is by assuming that whenever build dependencies change, their artifacts remain constant. That’s clearly wrong and any package that builds with dune should be rebuilt when dune is updated.

**Comment by andreypopp** [comment]
Thanks!

File: merlin-lsp.opam [hunk 1]
```diff
@@ -10,8 +10,7 @@ build: [
   ["dune" "runtest" "-p" name "-j" jobs] {with-test}
 ]
 depends: [
-  "ocaml" {>= "4.02.1" & < "4.09"}
```
Comment (anuragsoni) [inline merlin-lsp.opam]: was removing the ocaml dependency intentional?

**Comment by rgrinberg** [comment]
It was intentional. The lsp binary itself doesn’t prescribe any version of OCaml, the restriction is inherited from merlin itself.

**Pull request merged**